cpp/dolfin/io/XDMFFile.cpp

```diff
@@ -56,6 +85,73 @@ XDMFFile::XDMFFile(MPI_Comm comm, const std::string filename)
   // Flush datasets to disk at each timestep. Allows inspection of the
   // HDF5 file whilst running, at some performance cost.
   parameters.add("flush_output", false);
+
+  // Synchronize before filesystem check
+  _mpi_comm.barrier();
+  bool exists = boost::filesystem::exists(_filename);
+  // If XDMF XML file exists physically
+  if (exists)
+  {
+    // Load its XML structure into internal XML file
+    pugi::xml_parse_result result = _xml_doc->load_file(_filename.c_str());
```

Comment:
Should we read the XML on one process only? I can't imagine filesystems liking this with 10k+ processes.

Reply:
That makes more sense, I agree. But this is the past design. 
I agree, that the only process working with XML file on filesystem level should be root process, while information needed to read from HDF file like hdf filepath of dataset must be communicated through MPI. I'd rather make this change in separate PR.